### PR TITLE
Improved error processing and service naming

### PR DESCRIPTION
**Pull request by gechetspr**
https://spryker.atlassian.net/browse/FRW-9638

### CODE REVIEW

File: src/Spryker/Service/Opentelemetry/Instrumentation/SprykerInstrumentationBootstrap.php [hunk 1]
```diff
@@ -295,20 +298,32 @@ protected static function formatSpanName(Request $request): string
     public static function shutdownHandler(): void
     {
         $resourceName = ResourceNameStorage::getInstance()->getName();
+        $customParamsStorage = CustomParameterStorage::getInstance();
         if ($resourceName) {
+            if ($customParamsStorage->getAttribute(OpentelemetryMonitoringExtensionPlugin::ATTRIBUTE_IS_CONSOLE_COMMAND)) {
+                $resourceName = 'CLI ' . $resourceName  ;
```
Comment (annakotiuk) [inline src/Spryker/Service/Opentelemetry/Instrumentation/SprykerInstrumentationBootstrap.php:304]: Minor: use sprintf. Also remove extra ` ` $resourceName  ;

File: src/Spryker/Service/Opentelemetry/OpentelemetryService.php [hunk 2]
```diff
@@ -49,4 +57,19 @@ public function setResourceName(string $name): void
...
+     * {@inheritDoc}
+     *
+     * @api
+     *
+     * @param string $message
+     * @param \Throwable $exception
+     *
+     * @return void
+     */
+    public function setError(string $message, \Throwable $exception): void
```
Comment (annakotiuk) [inline src/Spryker/Service/Opentelemetry/OpentelemetryService.php:71]: Move Throwable to use block

**Review by dimitriyTsemma — DISMISSED** [review]

File: src/Spryker/Service/Opentelemetry/Plugin/OpentelemetryMonitoringExtensionPlugin.php [hunk 3]
```diff
@@ -112,15 +110,16 @@ public function markIgnoreTransaction(): void
 
     /**
      * Specification:
-     * - Opentelemetry has no attributes to specify background jobs. Service name already defined for CLI commands.
+     * - Adds custom param that will be added to root span.
+     * - Adds CLI prefix to the service name.
```
Comment (annakotiuk) [inline src/Spryker/Service/Opentelemetry/Plugin/OpentelemetryMonitoringExtensionPlugin.php:114]: Does it really add a prefix to service name? I suppose prefix is added in https://github.com/spryker/opentelemetry/pull/7/files#diff-2d42e6c1831b69e5df27557d3ac81a2350ffa520ccd357087ba1a0eeae11c1daR303

File: src/Spryker/Service/Opentelemetry/Instrumentation/SprykerInstrumentationBootstrap.php [hunk 4]
```diff
@@ -295,20 +298,32 @@ protected static function formatSpanName(Request $request): string
     public static function shutdownHandler(): void
     {
         $resourceName = ResourceNameStorage::getInstance()->getName();
+        $customParamsStorage = CustomParameterStorage::getInstance();
```
[the remark this answers is not in the document]
Comment (gechetspr) [reply src/Spryker/Service/Opentelemetry/Instrumentation/SprykerInstrumentationBootstrap.php:301]: Can't be. This class started before Spyker is booted and finishes after it's down, so no factories are available here

**Comment by annakotiuk** [comment]
Validations are red

File: src/Spryker/Service/Opentelemetry/Plugin/OpentelemetryMonitoringExtensionPlugin.php (hunk 3 above)
Comment (gechetspr) [reply src/Spryker/Service/Opentelemetry/Plugin/OpentelemetryMonitoringExtensionPlugin.php:114]: Well, yeah. The service do nothing with prefix, but the instrumentation code will use it after, so I would say it makes sense to mention it here for visibility

**Comment by gechetspr** [comment]
Validation is red and this is expected due to missing a proper PHP image with required extensions

File: src/Spryker/Service/Opentelemetry/Instrumentation/SprykerInstrumentationBootstrap.php (hunk 1 above)
Comment (gechetspr) [reply src/Spryker/Service/Opentelemetry/Instrumentation/SprykerInstrumentationBootstrap.php:304]: space is needed, but sprintf is a good idea, updating

**Comment by annakotiuk** [comment]
Is validation error here https://release.spryker.com/integration/view/spryker/opentelemetry/0c6136ae085629a584f575f01d60f0966f0f0492 expected?

File: src/Spryker/Service/Opentelemetry/Plugin/OpentelemetryMonitoringExtensionPlugin.php (hunk 3 above)
Comment (annakotiuk) [reply src/Spryker/Service/Opentelemetry/Plugin/OpentelemetryMonitoringExtensionPlugin.php:114]: Maybe than rephrase like 
Is used for adding CLI prefix to the service name ?

**Comment by gechetspr** [comment]
yes. Spryker CV validation red is also expected

**Review by annakotiuk — APPROVED** [review]

**Review by asmarovydlo — APPROVED** [review]

**Pull request merged**